timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py

```diff
@@ -350,22 +351,9 @@ def predict(
             df = self._gluonts_forecasts_to_data_frame(
                 predicted_targets,
                 quantile_levels=quantile_levels or self.quantile_levels,
+                data=data,
             )
 
-            # if index type is different than the input data, cast it back
-            if len(df.index.levels[0]) > 0:
-                prediction_index_type = type(df.index.levels[0][0])
-                if prediction_index_type is not input_index_type:
-                    df.set_index(
-                        df.index.set_levels([input_index_type(i) for i in df.index.levels[0]], level=0),
-                        inplace=True,
-                    )
-
-        # Make sure the item_ids are sorted in the same order as in data
-        df = df.loc[data.item_ids]
-        # GluonTS uses pd.Period internally, which may lead to loss of precision (e.g., "2020-01-01 12:00" becomes
-        # "2020-01-01 00:00"). We manually set the index to avoid such problems.
-        df.index = get_forecast_horizon_index_ts_dataframe(data, self.prediction_length)
```

Comment:
We now handle this problem inside `_gluonts_forecasts_to_data_frame`. This, however, requires us to pass `data` to the method to avoid the loss of timestamp precision.